Add TextStream dumping for WebKit::RepaintRequirements

aprotyas opened the pull request:
#### f54cee4c85ae983abccbed7b905890b358a15f73
<pre>
Add TextStream dumping for WebKit::RepaintRequirements
<a href="https://bugs.webkit.org/show_bug.cgi?id=274189">https://bugs.webkit.org/show_bug.cgi?id=274189</a>
<a href="https://rdar.apple.com/128097770">rdar://128097770</a>

Reviewed by Simon Fraser and Sammy Gill.

This patch typedefs OptionSet&lt;RepaintRequirement&gt; to RepaintRequirements
for easier reference, and adds a TextStream operator&lt;&lt; overload for said
type to aid with debugging.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::updateTrackedAnnotation):
(WebKit::UnifiedPDFPlugin::handlePDFActionForAnnotation):
(WebKit::AnnotationTrackingState::startAnnotationTracking):
(WebKit::UnifiedPDFPlugin::setDisplayModeAndUpdateLayout):
(WebKit::operator&lt;&lt;):

Canonical link: <a href="https://commits.webkit.org/278798@main">https://commits.webkit.org/278798@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0fbde6b1429206e34e942b638de8f970b1ecd3d5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51593 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30905 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3949 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54860 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2286 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37244 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1967 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41997 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53692 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28540 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44502 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23125 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25839 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1853 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56452 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26713 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1721 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49395 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27952 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44573 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/48594 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11291 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28847 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27688 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
```diff
@@ -626,6 +631,8 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
 #endif
 };
 
+WTF::TextStream& operator<<(WTF::TextStream&, const RepaintRequirements&);
```
No need to pass the 2nd arg by reference; it's tiny.

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/0fbde6b1429206e34e942b638de8f970b1ecd3d5

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -4154,6 +4155,36 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
 
+TextStream& operator<<(TextStream& ts, const RepaintRequirements& requirements)
+{
+    ts << '[';
```
TextStream knows how to dump OptionSets. You just need to define `TextStream& operator<<` for `RepaintRequirement`

aprotyas replied:
TIL, fixed in https://github.com/WebKit/WebKit/commit/0fbde6b1429206e34e942b638de8f970b1ecd3d5

aprotyas committed:
Add TextStream dumping for WebKit::RepaintRequirements
https://bugs.webkit.org/show_bug.cgi?id=274189
rdar://128097770

Reviewed by Simon Fraser and Sammy Gill.

This patch typedefs OptionSet<RepaintRequirement> to RepaintRequirements
for easier reference, and adds a TextStream operator<< overload for said
type to aid with debugging.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::updateTrackedAnnotation):
(WebKit::UnifiedPDFPlugin::handlePDFActionForAnnotation):
(WebKit::AnnotationTrackingState::startAnnotationTracking):
(WebKit::UnifiedPDFPlugin::setDisplayModeAndUpdateLayout):
(WebKit::operator<<):

Canonical link: https://commits.webkit.org/278798@main
